homeassistant/components/workday/config_flow.py

```diff
@@ -0,0 +1,198 @@
+"""Config flow for Workday."""
+import logging
+from types import MappingProxyType
+from typing import Any
+
+import holidays
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.const import CONF_COUNTRY, CONF_NAME
+from homeassistant.core import callback
+from homeassistant.data_entry_flow import FlowResult
+
+# from homeassistant.helpers import selector
+import homeassistant.helpers.config_validation as cv
+
+from . import util
+from .const import (
+    ALLOWED_DAYS,
+    CONF_ADD_HOLIDAYS,
+    CONF_EXCLUDES,
+    CONF_OFFSET,
+    CONF_PROVINCE,
+    CONF_REMOVE_HOLIDAYS,
+    CONF_WORKDAYS,
+    DEFAULT_EXCLUDES,
+    DEFAULT_OFFSET,
+    DEFAULT_WORKDAYS,
+    DOMAIN,
+)
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class WorkdayOptionsFlowHandler(config_entries.OptionsFlow):
+    """Handles the option flow for Workday."""
+
+    def __init__(self, entry: config_entries.ConfigEntry) -> None:
+        """Initialize Workday options flow."""
+        _LOGGER.debug(
+            "Initializing Workday options flow with ConfigEntry: %s",
+            util.config_entry_to_string(entry),
+        )
+        self._entry = entry
+
+    async def async_step_init(
+        self, user_input: dict[str, Any] | None = None
+    ) -> FlowResult:
+        """Handle user-initiated configuration."""
+        errors: dict[str, str] = {}
+        _LOGGER.debug(
+            "WorkdayOptionsFlowHandler.async_step_init user_input: %s", user_input
+        )
+
+        if user_input is not None:
+            try:
+                holiday_data = {
+                    CONF_NAME: self._entry.unique_id,
+                    CONF_COUNTRY: self._entry.data[CONF_COUNTRY],
+                }
+                holiday_data.update(self._entry.data)
+                holiday_data.update(user_input)
+                util.build_holidays(holiday_data)
+            except util.AddHolidayError:
+                errors[CONF_ADD_HOLIDAYS] = "bad_holiday"
+            except util.NoSuchHolidayError:
+                errors[CONF_REMOVE_HOLIDAYS] = "no_such_holiday"
+            else:
+                return self.async_create_entry(data=user_input)
+
+        return self._async_show_form(
+            step_id="init", user_input=user_input, errors=errors
+        )
+
+    @callback
+    def _async_show_form(
+        self,
+        step_id: str,
+        user_input: dict[str, Any] | MappingProxyType[str, Any] | None = None,
+        errors: dict[str, str] | None = None,
+    ) -> FlowResult:
+        if user_input is None:
+            _LOGGER.debug(
+                "No user_input, using default values: %s", self._entry.options
+            )
+            user_input = self._entry.options
+
+        return self.async_show_form(
+            step_id=step_id,
+            data_schema=vol.Schema(
```

Comment:
You should turn the schema into a constant, and then leverage `self.add_suggested_values_to_schema`.